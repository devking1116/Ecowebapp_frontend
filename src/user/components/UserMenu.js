/* @flow */
import React from 'react';
import { Icon } from 'semantic-ui-react';

import { HeaderMenu } from '../../core';
import UserMenuDropdown from './UserMenuDropdown';

type Props = {
  clientName: string,
  clientId: string,
  isAdmin?: bool,
  performLogout: () => void,
};

/**
 * The user menu that is in the header.
 */
export default function UserMenu(props: Props): React.Element<*> {
  const { clientName, clientId, isAdmin, performLogout, ...otherProps } = props;

  return (
    <HeaderMenu {...otherProps}>
      <Icon name="bug" size="big" {...otherProps} />
      <Icon name="alarm outline" size="big" {...otherProps} />
    {isAdmin?
      <Icon name="settings" size="big" {...otherProps} />:
      null
    }
      <Icon name="ellipsis horizontal" size="big" {...otherProps} />
      <UserMenuDropdown icon={<Icon name="user" size="big" {...otherProps} />} >
        <span style={{ display: 'flex', flexDirection: 'column' }}>
          <span>{clientName}</span>
          <span>Client ID: {clientId}</span>
        </span>
        <a onClick={performLogout}>
          Sign Out
        </a>
      </UserMenuDropdown>
    </HeaderMenu>
  );
}
