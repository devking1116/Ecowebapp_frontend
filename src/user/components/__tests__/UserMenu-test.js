import React from 'react';
import renderer from 'react-test-renderer';

import UserMenu from '../UserMenu';

test('it displays the client name and id.', () => {
  const component = renderer.create(
    <UserMenu clientName="Client Name" clientId="123456" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it only displays the Admin dropdown if isAdmin is set.', () => {
  const noAdminComponent = renderer.create(
    <UserMenu />
  );
  const adminComponent = renderer.create(
    <UserMenu isAdmin />
  );

  const noAdminTree = noAdminComponent.toJSON();
  const adminTree = adminComponent.toJSON();

  expect(noAdminTree).toMatchSnapshot();
  expect(adminTree).toMatchSnapshot();
});
