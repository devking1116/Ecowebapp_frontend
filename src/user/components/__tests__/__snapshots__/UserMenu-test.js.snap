exports[`test it displays the client name and id. 1`] = `
<div
  className="ui inverted secondary menu"
  style={
    Object {
      "height": "100%",
      "marginBottom": 0,
      "marginTop": 0,
    }
  }>
  <div
    className="item"
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="bug big icon" />
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="alarm outline big icon" />
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="ellipsis horizontal big icon" />
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <div
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "WebkitJustifyContent": "center",
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "center",
        }
      }>
      <i
        aria-hidden="true"
        className="user big icon" />
    </div>
  </div>
</div>
`;

exports[`test it only displays the Admin dropdown if isAdmin is set. 1`] = `
<div
  className="ui inverted secondary menu"
  style={
    Object {
      "height": "100%",
      "marginBottom": 0,
      "marginTop": 0,
    }
  }>
  <div
    className="item"
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="bug big icon" />
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="alarm outline big icon" />
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="ellipsis horizontal big icon" />
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <div
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "WebkitJustifyContent": "center",
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "center",
        }
      }>
      <i
        aria-hidden="true"
        className="user big icon" />
    </div>
  </div>
</div>
`;

exports[`test it only displays the Admin dropdown if isAdmin is set. 2`] = `
<div
  className="ui inverted secondary menu"
  style={
    Object {
      "height": "100%",
      "marginBottom": 0,
      "marginTop": 0,
    }
  }>
  <div
    className="item"
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="bug big icon" />
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="alarm outline big icon" />
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="settings big icon" />
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="ellipsis horizontal big icon" />
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <div
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "WebkitJustifyContent": "center",
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "center",
        }
      }>
      <i
        aria-hidden="true"
        className="user big icon" />
    </div>
  </div>
</div>
`;
